Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1771,6 +1773,20 @@ public JSONObject getLatestReferringParams() {
         return latestParams;
     }
 
+    private CountDownLatch latch = null;
+    public JSONObject getLatestReferringParamsSync() {
+        latch = new CountDownLatch(1);
+        try {
```

Comment:
Will this work when `getLatestParam` called from the same thread as we do internal initialization?
Internal init method is working UI thread (Since started from UI init methods). So It seems like this will prevent init happening when called from UI thread